app/src/main/java/io/apicurio/registry/logging/audit/AuditedInterceptor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.logging.audit;
+
+
+import org.slf4j.Logger;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Interceptor that executes around methods annotated with {@link Audited}
+ * <p>
+ * This interceptor follows the execution of a method and marks the audit entry as failed if the inner method throws an exception.
+ * <p>
+ * This interceptor reads the inner method parameters to gather extra information for the audit entry.
+ */
+@Audited
+@Interceptor
+@Priority(Interceptor.Priority.APPLICATION)
```

Comment:
This should probably be adjusted so the auditing interceptor is executed before e.g. `AuthorizedInterceptor`.